### PR TITLE
Simplify x86 msvc test

### PR DESCRIPTION
**Pull request by jon-turney**

### CODE REVIEW

**Comment by jon-turney** [comment]
This addresses the exception reported in #4419 in a different way.

Note that `detect_windows_arch()` is still possibly not quite right: Windows on ARM64 also has WOW32, so could run the output of a Windows/x86 targeted (cross-)compiler.

File: mesonbuild/environment.py [hunk 1]
```diff
@@ -624,11 +611,11 @@ def sanitize(p):
                 if version == 'unknown version':
                     version = search_version(out)
                 if version == 'unknown version':
-                    m = 'Failed to detect MSVC compiler arch: stderr was\n{!r}'
+                    m = 'Failed to detect MSVC compiler version: stderr was\n{!r}'
                     raise EnvironmentException(m.format(err))
-                is_64 = err.split('\n')[0].endswith(' x64')
+                arch = err.split('\n')[0].rsplit(' ', 1)[1].strip()
```
Comment (jon-turney) [inline mesonbuild/environment.py]: This is far too brittle. Compiler older than an certain vintage (2005?) possibly don't identify the target, so we should default to x86 (as we currently do) in that case.

**Comment by jpakkane** [comment]
LGTM.

**Comment by nioncode** [comment]
Ping. This already received a LGTM, so why hasn't it?

**Pull request merged**